### ISSUE TITLE
[Machine learning] Gradient Descent algorithm and its variants in Neural Network

### ISSUE THREAD

**Issue by vincentkims49** [section-engineering-education/engineering-education#4503]
#### Proposed title of article
Gradient Descent algorithm and its variants in Neural Network
#### Proposed article introduction
Gradient descent is a well-known optimization algorithm for training machine learning models and neural networks. These models learn over time with the help of training data, and the cost function within gradient descent acts as a barometer, gauging its accuracy with each iteration of parameter updates.
Gradient descent is the most commonly used algorithm for training neural networks. We'll define it later, but for now remember this: the gradient is a numeric calculation that tells us how to adjust the parameters of a network so that the output deviation is as small as possible.
#### Key takeaways 
- Types of gradient Descent
- Algorithm for stochastic gradient descent
- Convergence trends in different variants of Gradient Descents
- Algorithm for mini-batch gradient descent
#### Article quality
The article will be about how to calculate gradient descent in a neural network and the algorithms that use gradient descent.
#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)
